src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -88,7 +97,13 @@ public override void OnRemoveBehavior()
 
         if (affectedAgent.Team == affectorAgent.Team) // Team hit.
         {
-            return; // TODO: should penalize affector.
+            if (affectedAgent.MissionPeer != null)
+            {
+                var crpgPeer = affectedAgent.MissionPeer.GetNetworkPeer().GetComponent<CrpgPeer>();
+                _agentsThatGotTeamhitByCrpgPeer.TryAdd(crpgPeer, affectedAgent);

```

Comment:
It sounds dangerous to me to keep a reference to an agent that could be destroyed at any moment. It would be a potential crash to access some properties of an unknown agent.

I think it would be safer to save the hits here, that would also make the code more explicit.

Also it probably doesn't work for multiple lifes mode such as siege.

Reply:
Reworked whole structure to only work with own Classes and primitive Types